shared/components/application/application.component.js

```diff
@@ -0,0 +1,38 @@
+/*global module */
+import 'babel-polyfill';
+import React from 'react';
+import LayoutComponent from '../layout/layout.component';
+
+class ApplicationComponent extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      i18n: props.i18n
+    };
```

Comment:
Is this necessary - can we just pass props.i18n to `getChildContext` - just so variables don't proliferate?

https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html